cmd/kubeadm/app/util/config/upgradeconfiguration_test.go

```diff
@@ -104,3 +107,244 @@ func TestDocMapToUpgradeConfiguration(t *testing.T) {
+				if diff := cmp.Diff(got, tt.want, cmpopts.IgnoreFields(kubeadmapi.UpgradeConfiguration{}, "Timeouts")); diff != "" {
+					t.Errorf("LoadUpgradeConfigurationFromFile returned unexpected diff (-want,+got):\n%s", diff)
+				}
+			}
+		})
+	}
+}
+
+func TestDefaultedUpgradeConfiguration(t *testing.T) {
+	options := LoadOrDefaultConfigurationOptions{}
+	tests := []struct {
+		name                string
+		defaultversionedcfg *kubeadmapiv1.UpgradeConfiguration
+		want                *kubeadmapi.UpgradeConfiguration
+	}{
+		{
+			name:                "defaultversionedcfg is empty",
+			defaultversionedcfg: &kubeadmapiv1.UpgradeConfiguration{},
+			want: &kubeadmapi.UpgradeConfiguration{
+				Apply: kubeadmapi.UpgradeApplyConfiguration{
+					CertificateRenewal: ptr.To(true),
+					EtcdUpgrade:        ptr.To(true),
+				},
+				Node: kubeadmapi.UpgradeNodeConfiguration{
+					CertificateRenewal: ptr.To(true),
+					EtcdUpgrade:        ptr.To(true),
+				},
+			},
+		},
+		{
+			name: "defaultversionedcfg has part of fields configured",
+			defaultversionedcfg: &kubeadmapiv1.UpgradeConfiguration{
+				Apply: kubeadmapiv1.UpgradeApplyConfiguration{
+					CertificateRenewal: ptr.To(false),
+				},
+				Node: kubeadmapiv1.UpgradeNodeConfiguration{
+					EtcdUpgrade: ptr.To(false),
+				},
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: kubeadmapiv1.SchemeGroupVersion.String(),
+					Kind:       constants.UpgradeConfigurationKind,
+				},
+			},
+			want: &kubeadmapi.UpgradeConfiguration{
+				Apply: kubeadmapi.UpgradeApplyConfiguration{
+					CertificateRenewal: ptr.To(false),
+					EtcdUpgrade:        ptr.To(true),
+				},
+				Node: kubeadmapi.UpgradeNodeConfiguration{
+					CertificateRenewal: ptr.To(true),
+					EtcdUpgrade:        ptr.To(false),
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, _ := DefaultedUpgradeConfiguration(tt.defaultversionedcfg, options)
+			if diff := cmp.Diff(got, tt.want, cmpopts.IgnoreFields(kubeadmapi.UpgradeConfiguration{}, "Timeouts")); diff != "" {
+				t.Errorf("DefaultedUpgradeConfiguration returned unexpected diff (-want,+got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestLoadOrDefaultUpgradeConfiguration(t *testing.T) {
+	tmpdir, err := os.MkdirTemp("", "")
+	if err != nil {
+		t.Fatalf("Couldn't create tmpdir: %v", err)
+	}
+	defer func() {
+		if err := os.RemoveAll(tmpdir); err != nil {
+			t.Fatalf("Couldn't remove tmpdir: %v", err)
+		}
+	}()
+	filename := "kubeadmConfig"
+	filePath := filepath.Join(tmpdir, filename)
+	fileContents := dedent.Dedent(`
+				apiVersion: kubeadm.k8s.io/v1beta4
+				kind: UpgradeConfiguration
+			`)
+	err = os.WriteFile(filePath, []byte(fileContents), 0644)
+	if err != nil {
+		t.Fatalf("Couldn't write content to file: %v", err)
+	}
+
+	options := LoadOrDefaultConfigurationOptions{}
+
+	tests := []struct {
+		name                string
+		cfgPath             string
+		defaultversionedcfg *kubeadmapiv1.UpgradeConfiguration
```

Comment:
```suggestion
		cfg *kubeadmapiv1.UpgradeConfiguration
```

change it again like TestDefaultedUpgradeConfiguration please